lib/local_server.coffee

```diff
@@ -26,52 +29,74 @@ class Server
   ###
 
   start: (port, cb) ->
-    opts = @project.config.server ? {}
-    opts.log = false
-
-    if @project.config.env is 'development'
-      opts.write = content:
-        "<!-- roots development configuration -->
-        <script>var __livereload = #{@project.config.live_reload};</script>
-        <script src='/__roots__/main.js'></script>"
-      opts.cache_control = {'**': 'max-age=0, no-cache, no-store'}
-
-    app = charge(@project.config.output_path(), opts)
-
-    if @project.config.env is 'development'
-      app.stack.splice app.stack.length - 2, 0,
-        route: '/__roots__'
-        handle: serve_static(path.resolve(__dirname, 'browser'))
-
-    @server = app.start(port, cb)
+    # opts = @project.config.server ? {}
+
+    bs_options =
+      port: port
+      logLevel: 'silent'
+      server:
+        baseDir: @project.config.output_path()
+
+    if @project.config.browser then _.merge(bs_options, @project.config.browser)
+
+    # add charge middleware after merge to prevent errors
+    opts = @project.config.server or {}
+    middlewares = []
+
+    if opts.clean_urls
+      middlewares.push(charge.hygienist(@project.config.output_path()))
+    if opts.exclude
+      middlewares.push(charge.escapist(opts.exclude))
+    if opts.auth
+      middlewares.push(charge.publicist(opts.auth))
+    if opts.cache_control
+      middlewares.push(charge.archivist(opts.cache_control))
+    if opts.gzip
+      middlewares.push(charge.minimist(opts.gzip))
+    if opts.log
+      middlewares.push(charge.journalist(opts.log))
+    if opts.error_page
+      middlewares.push(charge.apologist(opts.error_page))
+
+    bs_options.server.middleware = middlewares
+
+    @bs.init(bs_options, cb)
 
   ###*
    * Close the server and remove it.
   ###
 
   stop: (cb) ->
-    @server.close(cb)
-    delete @server
+    @bs.exit()
 
   ###*
-   * Send a message through websockets to the browser.
-   *
-   * @param  {String} k - message key
-   * @param  {*} v - message value
+   * Reload the browser
   ###
 
-  send_msg: (k, v) ->
-    @server.send(type: k, data: v)
+  reload: ->
+    @bs.reload()
+
+  ###*
+   * Inject loading spinner while compiling
+  ###
+  compiling: ->
+    @bs.notify('<div id="roots-load-container"><div id="roots-compile-loader">
+    <div id="l1"></div><div id="l2"></div><div id="l3"></div><div id="l4"></div>
+    <div id="l5"></div><div id="l6"></div><div id="l7"></div><div id="l8"></div>
+    </div></div>')
```

Comment:
Just a formatting preference thing, but any reason you're not using CoffeeScript's awesome multiline strings for embedded HTML snippets?

```coffee
@bs.notify """
  <div id="roots-load-container">
    <div id="roots-compile-loader">
      <div id="l1"></div>
      <div id="l2"></div>
      <div id="l3"></div>
      <div id="l4"></div>
      <div id="l5"></div>
      <div id="l6"></div>
      <div id="l7"></div>
      <div id="l8"></div>
    </div>
  </div>
"""
```